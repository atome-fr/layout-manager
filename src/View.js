import React from 'react';

/**
 * Method for render the component
 * @param {Object} props The props passed to the component
 */
function View(props) {
    return (
        <div className="component">
            {props.children}
        </div>
    );
}

export default View;