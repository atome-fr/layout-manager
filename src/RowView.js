import React, {Component} from 'react';

// Import component
import View from './View';

/**
 * Class used to manage a row view
 * @class
 */
class RowView extends Component {

    /**
     * Method for render the components
     * @returns {*}
     */
    render() {
        return (
            <div className="row">
                <View>
                    {this.props.view.component}
                </View>
            </div>
        );
    }
}

export default RowView;