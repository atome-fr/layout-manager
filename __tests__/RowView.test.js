import React from 'react';
import RowView from '../src/RowView';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import View from "../src/View";

describe('RowView Component', () => {

    describe('UI Render Test', () => {

        it('should render with the component passed in parameter', () => {
            const wrapper = shallow(<RowView  view={{component: RowView}}/>);
            const tree = toJson(wrapper);

            expect(tree).toMatchSnapshot();
            expect(wrapper.find(View).length).toBe(1);
        });

    });

});