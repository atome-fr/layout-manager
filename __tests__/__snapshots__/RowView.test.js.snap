// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowView Component UI Render Test should render with the component passed in parameter 1`] = `
<div
  className="row"
>
  <View />
</div>
`;
